core/src/test/java/org/apache/iceberg/hadoop/TestCachingCatalog.java

```diff
@@ -123,4 +150,169 @@ public void testTableName() throws Exception {
+    metadataTables(tableIdent).forEach(metadataTbl -> assertCatalogEntryHasAge("The metadata tables should have an age of zero on load",
+        catalog, metadataTbl, Duration.ZERO));
+
+    // Move time forward now that metadata tables are loaded and then access the data table again, to reset its value.
+    ticker.advance(HALF_OF_EXPIRATION);
+    metadataTables(tableIdent).forEach(catalog::loadTable);
+    metadataTables(tableIdent).forEach(metadataTbl ->
+        assertCatalogEntryHasAge("Reloading the metadata tables should reset their age",
+            catalog, metadataTbl, Duration.ZERO));
+
+    // Sanity check. Because the table is still in the cache, its ttl isn't updated on metadata table load
+    assertCatalogEntryHasAge("Loading metadata tables should not reload origin table if its cached already",
+        catalog, tableIdent, HALF_OF_EXPIRATION);
+
+    // Move time forward the other half of the expiration interval
+    ticker.advance(HALF_OF_EXPIRATION);
+    Assertions.assertThat(catalog.cache().asMap()).doesNotContainKey(tableIdent);
+    assertCatalogHasExpiredMetadataTables(catalog, tableIdent);
+  }
+
+  // Move this method entirely in place.
+  public static void assertCatalogEntryHasAge(String assertionMessage, TestableCachingCatalog catalog,
+      TableIdentifier identifier, Duration expectedAge) {
+    Assertions.assertThat(catalog.cache().asMap()).containsKey(identifier);
+    Assertions.assertThat(catalog.ageOf(identifier))
+        .isPresent().get().isEqualTo(expectedAge);
+  }
+
+  public static void assertCatalogEntryHasTtl(
+      String assertionMessage, TestableCachingCatalog catalog, TableIdentifier ident, Duration expectedTimeLeft) {
+    Assert.assertEquals(assertionMessage,
+        Optional.ofNullable(expectedTimeLeft), catalog.remainingAgeFor(ident));
+  }
+
+  public static void assertMetadataTablesAreCached(TestableCachingCatalog catalog, TableIdentifier tableIdentifier) {
+    metadataTables(tableIdentifier).forEach(
+        metadataTable -> Assertions.assertThat(catalog.cache().asMap()).containsKey(metadataTable));
+  }
+
+  public static void assertCatalogHasExpiredMetadataTables(
+      TestableCachingCatalog catalog, TableIdentifier tableIdent) {
+    // Sanity check tha the table itself is expired
+    Assert.assertNull("The table should not be served by the CachingCatalog's cache",
+        catalog.tableFromCacheQuietly(tableIdent));
+    metadataTables(tableIdent)
+        .forEach(metadataTable ->
+            Assert.assertNull("The CachingCatalog should not return metadata tables for a TTL'd table",
+                catalog.tableFromCacheQuietly(metadataTable)));
+  }
+
+  public static List<TableIdentifier> metadataTables(TableIdentifier tableIdent) {
+    return Arrays.stream(MetadataTableType.values())
+        .map(type -> TableIdentifier.parse(tableIdent + "." + type.name().toLowerCase(Locale.ROOT)))
+        .collect(Collectors.toList());
+  }
```

Comment:
I'm going to continue to in-line most of these. I just wanted to get them out of the old class with the assertion helpers first and foremost.